libafl/src/stages/calibrate.rs

```diff
@@ -161,24 +193,40 @@ where
                 history_map.resize(map_len, O::Entry::default());
             }
 
-            for (first, (cur, history)) in
-                map_first.iter().zip(map.iter().zip(history_map.iter_mut()))
+            for (idx, (first, (cur, history))) in map_first
+                .iter()
+                .zip(map.iter().zip(history_map.iter_mut()))
+                .enumerate()
             {
                 if *first != *cur && *history != O::Entry::max_value() {
                     *history = O::Entry::max_value();
-                    unstable_entries += 1;
+                    unstable_entries.insert(idx);
```

Comment:
You could use a bool array of instead of a hashset for this step, should be a lot more cache friendly (hashmaps are not actually fast)

Reply:
(An array won't always be faster but for a large amount of elements it might be?)
Alternatively, just use a Vec here, it's getting inserted in a set later, anyway.
Other idea, we could benchmark BTreeSet vs HashSet

Reply:
Fwiw we can add micro benchmarks if we're interested, like so:
https://github.com/AFLplusplus/LibAFL/blob/2504b6dae3584f12b7ed1309919d6a5611278545/utils/libafl_benches/benches/rand_speeds.rs

Reply:
If we are worried about performance, then another option might be to only update the stats every N seconds, or N iterations. It is unlikely someone will need instantaneous updates?

Reply:
Calibration is done for every new testcase right now, there are other things associated with it

Reply:
I meant maybe just the bit for stability, rather than the whole thing. But I don’t know the architecture of LibAFL well enough to say whether that’s practical.

Reply:
ok i'll use vec